service/matching/task_queue_manager.go

```diff
@@ -718,6 +718,10 @@ func (c *taskQueueManagerImpl) trySyncMatch(ctx context.Context, params addTaskP
 	if params.forwardedFrom == "" && c.config.TestDisableSyncMatch() {
 		return false, nil
 	}
+	if c.config.FairOrdering() && c.taskReader.hasDispatchableTasks() {
```

Comment:
I see a tradeoff between aborting sync match immediately and allowing it to wait the usual 200ms (at a lower priority, using the two-channel approach):

- if we abort immediately, we return to history faster and so it can push tasks to matching faster, but we guarantee to push everything through the db
- if we wait, then there's a chance that within the 200ms, the last remaining backlog tasks could get dispatched and we could successfully sync match. how likely that is depends how big the typical backlog is.

So it seems like it depends whether we tend to have a small number of large bursts (i.e. we usually have no backlog but sometimes we have a big backlog), or a large number of small bursts (i.e. we have 1-2 tasks in the backlog somewhat often). I don't actually know the answer (and it depends on the workload) but I suspect it's closer to the latter.

That suggests to me that for the most control, we might want to support both,